Protocol/Tests/Protocol/QActions/QAction/CSharpCoreInterAppBrokerSupport.cs

```diff
@@ -179,37 +160,42 @@ private void CheckForSetParameter(CallingMethodClass callingMethod)
             {
-                results.Add(Error.InvalidInterAppReplyLogic(test, qAction, qAction, "Protocol", "SetParameter(9000001", qAction.Id.RawValue));
+                results.Add(Error.InvalidInterAppReplyLogic(test, qAction, qAction, "Protocol", "SetParameter(9000001", qAction.Id.RawValue)
+                                 .WithCSharp(callingMethod));
             }
         }
 
-        private bool IsArgumentPropertyFromReturnAddress(ISymbol symbol)
+        private static bool IsArgumentPropertyFromReturnAddress(ISymbol symbol)
         {
             foreach (var reference in symbol.DeclaringSyntaxReferences)
             {
                 var syntax = reference.GetSyntax();
-                if (syntax is VariableDeclaratorSyntax variableDeclarator)
+                if (!(syntax is VariableDeclaratorSyntax variableDeclarator))
+                {
+                    continue;
+                }
+
+                var initializer = variableDeclarator.Initializer;
+                if (!(initializer?.Value is MemberAccessExpressionSyntax memberAccess))
                 {
-                    var initializer = variableDeclarator.Initializer;
-                    if (initializer != null && initializer.Value is MemberAccessExpressionSyntax memberAccess)
+                    continue;
+                }
+
+                // At this point we found a declaration to a property called AgentId
+                if (memberAccess.Name.Identifier.Text.Equals("AgentId"))
+                {
+                    // Double-check the Class this property comes from is our ReturnAddress class.
+                    // This is currently performed using Syntax Analysis. As symbol parsing requires valid compilation.
+                    // Get the full text of the expression leading to 'AgentId'
+                    var instanceExpression = memberAccess.Expression.ToString();
+
+                    if (instanceExpression.EndsWith("ReturnAddress"))
                     {
-                        // at this point we found a declaration to a property called AgentId
-                        if (memberAccess.Name.Identifier.Text.Equals("AgentId"))
-                        {
-                            // Double check the Class this property comes from is our ReturnAddress class.
-                            // This is currently performed using Syntax Analysis. As symbol parsing requires valid compilation.
-                            // Get the full text of the expression leading to 'AgentId'
-                            var instanceExpression = memberAccess.Expression.ToString();
-
-                            if (instanceExpression.EndsWith("ReturnAddress"))
-                            {
-                                return true;
-                            }
-                        }
-
-                        return false;
+                        return true;
                     }
                 }
+
+                return false;
```

Comment:
Doesn't this only lets the foreach run once? If that is our goal then we should use .FirstOrDefault on the enumeration.
In the original code I was looping through all definitions to try and cover all bases, but I do also think there will always ever be a single one.

Reply:
As discussed at the desk, this return was in the original code, but can be removed.